src/screens/Wallets/components/Header.tsx

```diff
@@ -65,20 +63,21 @@ export function HomeHeader(): JSX.Element {
     const res = (await Cache.getItem(CacheKey.LastNotificationTimestamp)) || 0;
     const count = notifications?.filter((notification) => {
       return notification?.createdAt.getTime() > res;
-    });
-    setNewNotificationsCount(count?.length);
+    })?.length;
+    setNewNotificationsCount(count);
   }, [notifications]);
 
   useEffect(() => {
     handleNotificationsCheck();
-  }, [handleNotificationsCheck]);
+  }, [notifications]);
```

Comment:
I think we can also use appState as a dependence, what do you think ?

Reply:
I think we need to check it on both navigation and app state focus

Reply:
what do you mean?

Reply:
whenever the screen is focused, or app become foreground, we can call the `handleNotificationsCheck` function